### DISCUSSION TITLE
The contrast of greyed out habits is too low in night mode

### DISCUSSION THREAD

**Discussion by hiqua** [iSoron/uhabits#413] [Feature Requests] [answered]
I'm one of those users who use night mode for everything, at any time of the day.

Unfortunately Loop Habit Tracker seems to keep the same grey for e.g. habits that have not been completed yet.

This makes it hard to distinguish between those that have been completed in the previous days (grey tick), and those that should be completed (grey cross).

Please consider using a different colour in night mode, to make the distinction easier.

Thanks for Loop Habit Tracker!

**Comment by hiqua** [comment]
Following up from #719: @melocene it's a valid improvement suggestion, it's just not a software bug as a crash or anything unexpected. Any PR improving the current situation is welcome.

**Comment by hiqua** [comment]
Possible fix: commits in https://github.com/iSoron/uhabits/pull/777 along with https://github.com/iSoron/uhabits/pull/777#issuecomment-775637708.

**Comment by ghost** [comment]
@hiqua Thank you for notifying me of that PR.

After years of using and recommending this app I no longer due either for various reasons. A couple reasons are, I can no longer legitimately use the app due to the accessibility issues, and what I perceive to be a user hostile approach to managing this project by @iSoron which was brought to light by this issue. As an open source contributor and maintainer myself I do understand it is his choice though.

While I am glad to see someone trying to help with this, I don't recall seeing the option he mentioned for question marks when I used the app so I am not sure if the PR will result in the same outcome with his suggestion added.

**Reply by iSoron** [reply]
@melocene We have explicitly acknowledged that this is a valid concern, and I am willing to merge any PRs improving the situation. The linked PR was closed due to inactivity, not because we are not willing to consider it.

**Reply by qbarron73** [reply]
> 
> 
> @melocene We have explicitly acknowledged that this is a valid concern, and I am willing to merge any PRs improving the situation. The linked PR was closed due to inactivity, not because we are not willing to consider it.

came to report same for light mode. i see a pr which got rejected for no reason so its untrue to say you are willing to merge prs to help. be honest with users pls

**Comment by iSoron** [comment, marked as answer]
Fixed in ee462715de625c37c6b785a78d80dd989368e3c3, and should become available in Loop 2.0.

**Reply by ghost** [reply]
@iSoron Thanks for trying to fix this, but locking my PR (https://github.com/iSoron/uhabits/pull/848) was not necessary, it was an attempt to work with you to get this resolved. In #777 someone did respond to your proposal and based on that point in time it looks like it would not have helped. I can however review your commit if you are interested in making this more accessible/usable for visually impaired users and let you know if it helped.

As far as adding options, it seems version 2 adds a few anyway, and having options for accessibility is a good thing, it adds flexibility since users can set things that better fit their situations rather than excluding them or forcing users to fit a specific mold.

Edit: After thinking on this further additional app options will be needed to accommodate those with various disabilities or impairments since needs will vary. Are you able to include documentation in this repository for future contributors telling them what is or isn't acceptable for improving the accessibility of this app please, or if these contributions will even be considered? This will help contributors make the best use of their time.

**Reply by iSoron** [reply]
Suggestions for improving accessibility are very welcome. Since these are proposing changes to the theme, I would like to see a proposed mockup before reviewing any code. This is already documented in `docs/GUIDELINES.md`:
> If your pull request involves changes to the user interface, please work on a mockup first and submit a draft pull request with your proposed UI changes before writing the code to make it functional, to gather feedback from other developers and users. Inkscape and Figma are good tools that you can use.

Whoever submits the PR should also be willing to make substantial changes to it.

**Reply by ghost** [reply]
Can you let me know what was missing from my PR? I did provide screenshots of the new option in the UI, and its effects on the main UI (with appropriate labels) which was otherwise unchanged. Possibly you wanted a screenshot of the whole settings menu instead? It also seemed like other users were fine with the look based on previous PR comments and reactions. #777 also had a screenshot with the similar change, and the proposed change.

To clarify though, were mockups of https://github.com/iSoron/uhabits/commit/ee462715de625c37c6b785a78d80dd989368e3c3  shared anywhere for user feedback per `docs/GUIDELINES.md` to allow discussion as to if it solves they visibility issue? At least when I look at the commit I am not seeing any, but perhaps they are elsewhere.

**Reply by iSoron** [reply]
> Can you let me know what was missing from my PR?

This is already clearly stated in the PR itself.

> To clarify though, were mockups of ee46271 shared anywhere for user feedback per docs/GUIDELINES.md to allow discussion as to if it solves they visibility issue?

This specific guideline is for submitting pull requests. Project maintainers can gather direct feedback from tens of thousands of users by releasing a new beta version of the app.
